pkgs/os-specific/linux/zfs/stable.nix

```diff
@@ -20,9 +20,7 @@ callPackage ./generic.nix args {
     then kernel.kernelOlder "6.8"
     else kernel.kernelOlder "6.2";
 
-  latestCompatibleLinuxPackages = if stdenv'.isx86_64 || removeLinuxDRM
-    then linuxKernel.packages.linux_6_7
-    else linuxKernel.packages.linux_6_1;
```

Comment:
I think this broke my setup. I have `6.1` on an aarch64 host on `23.11`. nixpkgs got updated last night: 

```
    • Updated input 'nixpkgs':
        'github:NixOS/nixpkgs/1487bdea619e4a7a53a4590c475deabb5a9d1bfb' (2024-04-03)
      → 'github:NixOS/nixpkgs/e38d7cb66ea4f7a0eb6681920615dfcc30fc2920' (2024-04-06)
```

https://github.com/NixOS/nixpkgs/compare/1487bdea619e4a7a53a4590c475deabb5a9d1bfb..e38d7cb66ea4f7a0eb6681920615dfcc30fc2920


`fra1-a` has `pkgs.zfs.latestCompatibleLinuxPackages` and now I see the following:

```
[motiejus@fwminex:~/code/config]$ nom build .#nixosConfigurations.fra1-a.config.system.build.toplevel
Using saved setting for 'extra-experimental-features = nix-command flakes' from ~/.local/share/nix/trusted-settings.json.
Using saved setting for 'trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=' from ~/.local/share/nix/trusted-settings.json.
Using saved setting for 'trusted-substituters = https://cache.nixos.org/' from ~/.local/share/nix/trusted-settings.json.
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'nixos-system-fra1-a-23.11.20240406.e38d7cb'
         whose name attribute is located at /nix/store/6nfpbajyzwxq6wrhn20l13rhckhv26rc-source/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'buildCommand' of derivation 'nixos-system-fra1-a-23.11.20240406.e38d7cb'

         at /nix/store/6nfpbajyzwxq6wrhn20l13rhckhv26rc-source/nixos/modules/system/activation/top-level.nix:53:5:

           52|     passAsFile = [ "extraDependencies" ];
           53|     buildCommand = systemBuilder;
             |     ^
           54|

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Package ‘zfs-kernel-2.2.3-6.6.25’ in /nix/store/6nfpbajyzwxq6wrhn20l13rhckhv26rc-source/pkgs/os-specific/linux/zfs/generic.nix:213 is marked as broken, refusing to evaluate.

       a) To temporarily allow broken packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_BROKEN=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowBroken = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowBroken = true; }
       to ~/.config/nixpkgs/config.nix.
┏━ 1 Errors: 
┃ error:
┃        … while calling the 'derivationStrict' builtin
┃ 
┃          at /builtin/derivation.nix:9:12: (source not available)
┃ 
┃        … while evaluating derivation 'nixos-system-fra1-a-23.11.20240406.e38d7cb'
┃          whose name attribute is located at /nix/store/6nfpbajyzwxq6wrhn20l13rhckhv26rc-source/pkgs/stdenv/generic/make-derivation.nix:348:7
┃ 
┃        … while evaluating attribute 'buildCommand' of derivation 'nixos-system-fra1-a-23.11.20240406.e38d7cb'
┃ 
┃          at /nix/store/6nfpbajyzwxq6wrhn20l13rhckhv26rc-source/nixos/modules/system/activation/top-level.nix:53:5:
┃ 
┃            52|     passAsFile = [ "extraDependencies" ];
┃            53|     buildCommand = systemBuilder;
┃              |     ^
┃            54|
┃ 
┃        (stack trace truncated; use '--show-trace' to show the full trace)
┃ 
┃        error: Package ‘zfs-kernel-2.2.3-6.6.25’ in /nix/store/6nfpbajyzwxq6wrhn20l13rhckhv26rc-source/pkgs/os-specific/linux/zfs/generic.nix:213 is marked a…
┃ 
┃        a) To temporarily allow broken packages, you can use an environment variable
┃           for a single invocation of the nix tools.
┃ 
┃             $ export NIXPKGS_ALLOW_BROKEN=1
┃ 
┃           Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
┃                 then pass `--impure` in order to allow use of environment variables.
┃ 
┃        b) For `nixos-rebuild` you can set
┃          { nixpkgs.config.allowBroken = true; }
┃        in configuration.nix to override this.
┃ 
┃        c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
┃          { allowBroken = true; }
┃        to ~/.config/nixpkgs/config.nix.
┣━━━                                                             
┗━ ∑ ⚠ Exited with 1 errors reported by nix at 23:59:20 after 22s
```

On 1487bdea619e4a7a53a4590c475deabb5a9d1bfb it worked fine. I believe this is because this PR drops 6.1 on 23.11.

Reply:
Also, relatedly: should we stick to the same minor kernel version on a "stable" NixOS release?

Reply:
The underlying issue you probably have is documented here: https://nixos.org/manual/nixos/stable/#sec-linux-zfs
Basically, as soon as kernels are EOL, they will be removed.
It's usually the best to stick to a kernel with longterm support when using ZFS.

Reply:
My host is running 6.1, which is LTS. This commit bumps it to 6.6, which is not supported by OpenZFS on arm64.

And according to the docs, `boot.kernelPackages = pkgs.zfs.latestCompatibleLinuxPackages;` should work?

Reply:
Oh I see what you mean, apologies!
Looks like the issue got fixed on master though. Will look tomorrow what exactly we need to backport.

Reply:
Thanks, that's great!

I created https://github.com/NixOS/nixpkgs/issues/302432 for visibility, so other possibly affected folks can look up the recent issues and find it easily.